rust-symcrypt/src/rsa/pkcs1.rs

```diff
@@ -154,7 +154,10 @@ impl RsaKey {
                 0,
             ) {
                 symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => Ok(()),
-                err => Err(err.into()),
+                err => Err(match err.into() {
+                    SymCryptError::InvalidArgument => SymCryptError::SignatureVerificationFailure,
```

Comment:
>SymCryptError::InvalidArgument => SymCryptError::SignatureVerificationFailure [](http://example.com/codeflow?start=20&length=77)

Why map this? We distinguish these error cases on purpose

Reply:
We are getting the InvalidArgument instead of SignatureVerificationFailure when passing in random signature bytes. From the caller's point of view we don't know why you are returning a different error?

Reply:
It is useful to expose the distinction to the caller as _normally_ `SYMCRYPT_INVALID_ARGUMENT` would indicate a programmer error, and `SYMCRYPT_SIGNATURE_VERIFICATION_FAILURE` would indicate bad data.

In general `SYMCRYPT_INVALID_ARGUMENT` indicates something wrong about how the function was invoked (i.e. clearly bad arguments like flags set incorrectly, length of signature not matching the RSA key, etc.), and `SYMCRYPT_SIGNATURE_VERIFICATION_FAILURE` indicates that the input looks reasonable, but the signature does not verify.

I believe that you would currently hit `SYMCRYPT_INVALID_ARGUMENT` if a signature is going to fail regardless of the hash value you're trying to verify (i.e. if the signature value > RSA modulus then the signature is always invalid for that RSA key). This could typically happen when mixing up which RSA key a signature should be verified with; but it could also happen due to a corrupted signature.

Perhaps in your tests you could distinguish by signing a random hash, and then attempting to verify a perturbed hash with the same signature; this should always return `SYMCRYPT_SIGNATURE_VERIFICATION_FAILURE`.